File: app/src/main/java/com/example/sekunda/Data/Day.java
package com.example.sekunda.Data;

import android.content.Context;

import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Calendar;
import java.util.Locale;

class Day {

    private ArrayList<Business> mBusinessArrayList;
    private Calendar mCalendar;
    private int mAllSeconds;

    Day(Calendar calendar, Context context) {
        mCalendar = calendar;
        mBusinessArrayList = getFulledArrayList(context);
        mAllSeconds = findTotalTime();
    }

    private int findTotalTime(){
        int sum = 0;
        for(Business business: mBusinessArrayList){
            sum += business.getSeconds();
        }
        return sum;
    }

    String getAllTime(){
        String ans;
        StringBuilder stringBuffer = new StringBuilder();
        int hours = this.mAllSeconds / 3600;
        int minutes = this.mAllSeconds / 60 - hours * 60;
        int seconds = this.mAllSeconds % 60;
        if (hours > 0){
            stringBuffer.append(hours).append("h");
        }
        if (minutes >= 0) stringBuffer.append(minutes>9?minutes:"0"+ minutes).append("m:");
        if (seconds >= 0) stringBuffer.append(seconds).append("s");

        ans = stringBuffer.toString();
        return ans;
    }

    ArrayList<Business> getBusinessArrayList() {
        return mBusinessArrayList;
    }

    String getCalendarTime(){
        SimpleDateFormat simpleDateFormat = new SimpleDateFormat(BusinessSQLiteOpenHelper.SHORT_PATTERN, Locale.ENGLISH);
        return simpleDateFormat.format(this.mCalendar.getTime());
    }

    private ArrayList<Business> getFulledArrayList(Context context){
        BusinessSQLiteOpenHelper sqLiteOpenHelper = new BusinessSQLiteOpenHelper(context);
        ArrayList<Business> businessArrayList = sqLiteOpenHelper.getFulledList(mCalendar);
        sqLiteOpenHelper.close();
        return businessArrayList;
    }

}

File: app/src/main/java/com/example/sekunda/data/Day.kt
package com.example.sekunda.data

import com.example.sekunda.SeKaundaApplication
import java.text.SimpleDateFormat
import java.util.*
import kotlin.collections.ArrayList

class Day(private val mCalendar: Calendar, val businessArrayList: ArrayList<Business>) {

    private val mAllSeconds: Int

    private fun findTotalTime(): Int {
        var sum = 0
        for (business in businessArrayList) {
            sum += business.seconds
        }
        return sum
    }

    val hmsTime: String
        get() {
            val ans: String
            val stringBuffer = StringBuilder()
            val hours = mAllSeconds / 3600
            val minutes = mAllSeconds / 60 - hours * 60
            val seconds = mAllSeconds % 60
            if (hours > 0) {
                stringBuffer.append(hours).append("h")
            }
            if (minutes >= 0) stringBuffer.append(if (minutes > 9) minutes else "0$minutes").append("m:")
            if (seconds >= 0) stringBuffer.append(seconds).append("s")
            ans = stringBuffer.toString()
            return ans
        }
    val dmyTime: String
        get() {
            val simpleDateFormat = SimpleDateFormat(Business.DMY_PATTERN, Locale.ENGLISH)
            return simpleDateFormat.format(mCalendar.time)
        }

    fun addEmptyBusiness() {
        businessArrayList.add(Business("Nothing Today", 0))
    }

    init {
        mAllSeconds = findTotalTime()
    }
}